easybuild/tools/filetools.py

```diff
@@ -154,22 +157,28 @@ def read_file(path, log_error=True):
             return None
 
 
-def write_file(path, txt, append=False):
+def write_file(path, txt, append=False, forced=False):
     """Write given contents to file at given path (overwrites current file contents!)."""
-    f = None
+
+    # early exit in 'dry run' mode
+    if not forced and build_option('extended_dry_run'):
+        dry_run_msg("file written: %s" % path, silent=build_option('silent'))
+        return
+
+    handle = None
     # note: we can't use try-except-finally, because Python 2.4 doesn't support it as a single block
```

Comment:
well, using `with` (context manager) avoids the need for `try-except-finally` all together, so let me use that instead :)